bmeg_app/components/func_repurpose_drug.py

```diff
@@ -0,0 +1,177 @@
+        q = G.query().V('Project:CCLE').out("cases").as_("p").out("samples").out("aliquots").out("drug_response").as_("dr").out("compounds").as_("c")
+        drugRes = q.aggregate(gripql.term("drugs", '_data.synonym')).execute()
+        drug_opts = [a['key'].upper() for a in drugRes[0]['drugs']['buckets']]
+        drug_opts = {}
+        for a in drugRes[0]['drugs']['buckets']:
+            drug_opts[a['key'].upper()]= a['key']
+        return drug_opts
+    #elif other project...load project specific properties
+def mappings_drugResp(selected_project):
+    if 'CCLE' == selected_project:
+        return {
+        'AAC':'$dr._data.aac',
+        'IC50':'$dr._data.ic50',
+        'EC50': '$dr._data.ec50'
+         }
+    #elif other project...load project specific properties
+
+def drugDetails(drugs_list):
+    '''
+    create table of drugs and their taxon.
+    '''
+    a=[]
+    b=[]
+    c=[]
+    d=[]
+    e=[]
+    f=[]
+    g=[]
+    for DRUG in drugs_list:
+        q=G.query().V().hasLabel('Compound').has(gripql.eq('_data.synonym', DRUG))
+        q=q.render(['_data.synonym','_data.pubchem_id','_data.taxonomy.direct-parent','_data.taxonomy.kingdom','_data.taxonomy.superclass','_data.taxonomy.class','_data.taxonomy.subclass','_data.taxonomy.description'])
+        for row in q:
+            a.append(row[0])
+            b.append(row[1])
+            c.append(row[2])
+            d.append(row[3])
+            e.append(row[4])
+            f.append(row[5])
+            g.append(row[6])
+    df = pd.DataFrame(list(zip(a,b,c,d,e,f,g)),columns=['Drug','PubChem ID','Direct Parent','Kingdom','Superclass','Class','Subclass'])
+    return df
+
+def counting(vals_list):
+    '''
+    ex. counting(res['Gender'])
+    '''
+    lab=[]
+    cts=[]
+    for k,v in Counter(vals_list).items():
+        lab.append(k)
+        cts.append(v)
+    return lab, cts
+
+def get_histogram_normal(data):
+    '''
+    input values to plot. can be pandas df['col']
+    yticks == how far apart ticks
+     returns go figure
+    '''
+    import pandas as pd
+    import plotly.graph_objects as go
+    fig = go.Figure(data=[go.Histogram(
+        x=data,
+        #marker=dict(color=box_color)
+        )]).update_xaxes(categoryorder="total descending")
+    fig.update_layout(margin={'t':0, 'b':0},
+        height=200,
+        yaxis=dict(tickmode='linear'),
+        #plot_bgcolor='white'
+    )
+    fig.update_xaxes(showticklabels=False)
+    fig.update_yaxes(showticklabels=False)
+    return fig
+    
+def piecharts_celllines(drug1,df, proj):
+    # Grab all cell lines for that selected drug and look up metadata
+    # samples = df[df['Drug']==drug1]['Cell Line'].dropna()
+    samples = df[df['Drug']==drug1].dropna(subset=['Drug Response'])['Cell Line']
+    sample=[]
+    a=[]
+    b=[]
+    c=[]
+    for s in samples:
+        q=G.query().V().hasLabel('Case').has(gripql.eq('$._data.case_id', s)).has(gripql.eq('$._data.project_id', proj)).as_('c')
```

Comment:
It would be faster to store the GID and access the data using that. If you want to display the Case_id to the user, store it, and translate to that before rendering to the screen, but in the code, you need to use the Gid so the records can be accessed directly. By accessing them using case_id, the database is forced to scan all the records to find a match.

Reply:
Updated to pass GID list through query to utilize indexing